### DISCUSSION TITLE
Federation questions

### DISCUSSION THREAD

**Discussion by h-2** [pixelfed/pixelfed#5113] [Q&A]
I have just "moved" one of my Mastodon accounts to Pixelfed, and have some questions regarding interaction with Mastodon accounts:

1. I don't seem to have reposts in my Homefeed of people that I follow. Is this always the case, or is this just for Mastodon-users I follow? Can I change this behaviour somewhere?
2. Sometimes Pixelfed reports that a post has N comments, but it won't show all of them. See an example below. This is both true for the total number and also for nested replies, where it might say that there are two but then only show one. Fedilab shows the exact same comments, but reports differently (only top-level comments are counted, and there are no reply-counts to replies, so I can't compare).

![image](https://github.com/pixelfed/pixelfed/assets/2996863/06f0e79f-f3ee-405d-bf37-996b7b2ba255)

**Comment by mitexleo** [comment]
> 1. I don't seem to have reposts in my Homefeed of people that I follow. Is this always the case, or is this just for Mastodon-users I follow? Can I change this behaviour somewhere?

It's same for all users not only Mastodon . It's WIP. Will be fixed in this weekend I hope.

**Comment by SteveDinn** [comment]
I am not seeing media-containing posts (not re-posts) from Mastodon users that I follow from my Pixelfed account. Is this related (i.e., a "federation issue") or expected behaviour?

**Comment by mitexleo** [comment]
> I am not seeing media-containing posts (not re-posts) from Mastodon users that I follow from my Pixelfed account. Is this related (i.e., a "federation issue") or expected behaviour?

No ... Looks like an issue with your instance!

**Comment by SteveDinn** [comment]
Yeah, my problem must be something different and weird. I noticed that I only have that problem with *some* accounts, but not all.

**Comment by Moonstone3010** [comment]
For testing, I created several accounts on different Pixelfed instances. I found an account with a large number of image posts on Mastodon and displayed this account sequentially on each Pixelfed instance on which I am logged in. The effect is that I see different content from each account. I've searched for information on how all this federation stuff should work, but got nothing but headaches.
Could someone at least try to explain why I see different posts of the same account from each instance?

**Comment by mitexleo** [comment]
> For testing, I created several accounts on different Pixelfed instances. I found an account with a large number of image posts on Mastodon and displayed this account sequentially on each Pixelfed instance on which I am logged in. The effect is that I see different content from each account. I've searched for information on how all this federation stuff should work, but got nothing but headaches. Could someone at least try to explain why I see different posts of the same account from each instance?

It depends on when the pixelfed server was federated with the for example Mastodon server. Pixelfed doesn't display older posts that were created before federating as far as I'm aware.

**Comment by Moonstone3010** [comment]
> It depends on when the pixelfed server was federated with the for example Mastodon server. 

Thanks for the info, but this only explains one out of five cases tested. 

- On `pixelfed.social` I see ~80 posts, even though the profile info says there are 211
- On `pixey.org` I see 16 posts, while the profile info says there are 29
- On `pxlmo.com` I see 4 posts, profile info says 17
- On `metapixl.com` also 4 posts (different from on pxlmo), and the profile info says 6
- On `pixel.infosec.exchange` there are no posts and this case is clear, because the instance was created later than the last post of this user

If it was about the date of federation then from date x all posts would be visible. This is not the case, they appear to be random.



> Pixelfed doesn't display older posts that were created before federating as far as I'm aware.

I would very much like to understand this. What is the point of creating new instances if they cannot see past posts? 
I think most people decide to follow someone based on that. If you create an account on a fresh instance then you will never see them. It follows that to access all posts it is best to create an account on the oldest one. In this situation what overall sense does decentralisation make?

**Comment by h-2** [comment]
> If it was about the date of federation then from date x all posts would be visible. This is not the case, they appear to be random.

I am also definitely not seeing all the posts that I expect to see in pixelfed. This certainly seems like a bug to me, and beside the performance issues (#4367), is one of the major pains for me at the moment.

> I would very much like to understand this. What is the point of creating new instances if they cannot see past posts?

I understand this problem and agree it is not what most people expect, but I think it is a separate issue from all the other ones, and I am not sure Pixelfed can do anything about it. AFAIK, ActivityPub is built as a push-infrastructure and not as a pull-infrastructure. This means, that unless you want all instances to always push all information to all other instances (lots of data, how do you know who to send to?); the subscribing to someone posts means getting everything from the point in time onwards.

There are certain things I could think of to still make the UX better, e.g. when you click on a person's profile, instead of showing whatever is cached on your instance, directly redirect to the person's profile on their instance (so you see everything). Unfortunately, on that person's instance, you have no way of following/interacting, but maybe some kind of overlay could be added or the remote instance be embedded in a frame or so, to still allow this.

**Reply by CptCurk** [reply]
Okay, thank you a lot. I understand why I don't see anything on my proper instance. Just need to wait the "push" moment. 
I'd love to be able to "follow" instances and having content on my instance. For now, all is empty ! I hope it's just a matter of time, as my server runs on a Rasp4, it can be long. 
But anyway, there is a lack of information about that. It's like two hours I'm looking for that. Peertube had it almost automaticly on the same Raspi.
